Fix validateWarning Story

juliawegmayr opened the pull request:
## Description

## Acceptance criteria

-   [ ] I have verified if my change requires [an example](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#example)
-   [ ] I have verified if my change requires [a changeset](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#changeset)
-   [ ] I have verified if my change requires [screenshots/screencasts](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#screenshotsscreencasts)

## Screenshots/screencasts

| Before | After |
| ------ | ----- |
| Link   | Link  |

## Open TODOs/questions

-   [ ] Add changeset

## Further information

-   Task: https://vivid-planet.atlassian.net/browse/COM-XXX

jamesricky reviewed (changes requested):
The story/usage is not the issue here, it's most likely something in the implementation of `Field` or it's validation. 

The example from the task, for instance has nothing to do with the story and also doesn't work:

```tsx
<TextField label="Name (error)" name="name-error" validate={() => "This is an error"} />
<TextField label="Name (warning)" name="name-warning" validateWarning={() => "This is a warning"} />
```

<img width="412" alt="image" src="https://github.com/user-attachments/assets/de89c027-f1a6-4418-b7aa-cd851d16bfe8" />

johnnyomair commented on storybook/src/docs/form/Validation.stories.tsx:
```diff
@@ -116,10 +116,17 @@ export const Warnings = () => {
             onSubmit={() => {
                 // noop
             }}
-            validateWarning={validateWarning}
```
Yes, we shouldn't remove form-level warnings.

juliawegmayr commented:
> The story/usage is not the issue here, it's most likely something in the implementation of `Field` or it's validation.
> 
> The example from the task, for instance has nothing to do with the story and also doesn't work:
> 

I couldn't reproduce this issue (not in the demo and also not in the storybook). I suggest to check it again as soon as there is an explicit case, where this does not work correctly.

> If this is valid, the email doesn't seem to be validated 🤔

This came up while testing the story/usage. I found out, that the form validation resets the displayed warning message from the form validation - although the validation itself worked. I fixed this now and adapted the PR description. I also reverted my changes in the story (I initially thought, that form and field validation should not be used at the same time).

johnnyomair commented on packages/admin/admin/src/FinalForm.tsx:
```diff
@@ -162,6 +162,8 @@ export function FinalForm<FormValues = AnyObject, InitialFormValues = Partial<Fo
 
         const registeredFields = formRenderProps.form.getRegisteredFields();
 
+        const formWarningsRef = useRef<Record<string, string | undefined>>({});
```
Naming: formLevelWarnings (ref can be omitted IMO)

johnnyomair commented on .changeset/famous-pianos-happen.md:
```diff
@@ -0,0 +1,5 @@
+"@comet/admin": patch
+---
+
+Fix display of warning messages for Forms that use both form- and field validation
```
```suggestion
Fix display of warnings for forms that use both form-level and field-level validation
```

johnnyomair commented on packages/admin/admin/src/FinalForm.tsx:
```diff
@@ -177,11 +179,16 @@ export function FinalForm<FormValues = AnyObject, InitialFormValues = Partial<Fo
                     if (!validationWarnings) {
                         registeredFields.forEach((fieldName) => {
-                            setFieldData(fieldName, { warning: undefined });
+                            if (formWarningsRef.current[fieldName]) {
+                                setFieldData(fieldName, { warning: undefined });
+                            }
```
This was hard to understand. We should either create a dedicated variable `hasFormLevelWarning` or add a comment explaining that this prevents resetting field-level warnings.
